test/e2e/shared/aws.go

```diff
@@ -692,3 +692,455 @@ func ListVPC(e2eCtx *E2EContext) int {
 
 	return len(out.Vpcs)
 }
+
+func GetVPC(e2eCtx *E2EContext, vpcID string) (*ec2.Vpc, error) {
+	ec2Svc := ec2.New(e2eCtx.AWSSession)
+
+	filter := &ec2.Filter{
+		Name:   aws.String("vpc-id"),
+		Values: aws.StringSlice([]string{vpcID}),
+	}
+
+	input := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			filter,
+		},
+	}
+
+	result, err := ec2Svc.DescribeVpcs(input)
+	if err != nil {
+		return nil, err
+	}
+	return result.Vpcs[0], nil
+}
+
+func CreateVPC(e2eCtx *E2EContext, vpcName string, cidrBlock string) (*ec2.Vpc, error) {
+	ec2Svc := ec2.New(e2eCtx.AWSSession)
+
+	input := &ec2.CreateVpcInput{
+		CidrBlock: aws.String(cidrBlock),
+		TagSpecifications: []*ec2.TagSpecification{
+			{
+				ResourceType: aws.String("vpc"),
+				Tags: []*ec2.Tag{
+					{
+						Key:   aws.String("Name"),
+						Value: aws.String(vpcName),
+					},
+				},
+			},
+		},
+	}
+	result, err := ec2Svc.CreateVpc(input)
+	if err != nil {
+		return nil, err
+	}
+	return result.Vpc, nil
+}
+
+func DeleteVPC(e2eCtx *E2EContext, vpcID string) (bool, error) {
+	ec2Svc := ec2.New(e2eCtx.AWSSession)
+
+	input := &ec2.DeleteVpcInput{
+		VpcId: aws.String(vpcID),
+	}
+	if _, err := ec2Svc.DeleteVpc(input); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func GetSubnet(e2eCtx *E2EContext, subnetID string) (*ec2.Subnet, error) {
+	ec2Svc := ec2.New(e2eCtx.AWSSession)
+
+	filter := &ec2.Filter{
+		Name:   aws.String("subnet-id"),
+		Values: aws.StringSlice([]string{subnetID}),
+	}
+
+	input := &ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			filter,
+		},
+	}
+
+	result, err := ec2Svc.DescribeSubnets(input)
+	if err != nil {
+		return nil, err
+	}
+	return result.Subnets[0], nil
+}
+
+func CreateSubnet(e2eCtx *E2EContext, subnetName string, cidrBlock string, az string, vpcID string) (*ec2.Subnet, error) {
```

Comment:
We add [CCM tags automatically](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2584) to unmanaged Subnets, in case want to make this more similar. Not required for this PR. Just FYI.

In summary, to identify a cluster's subnets, CCM looks for the following tags:​

- kubernetes.io/cluster/YOUR-CLUSTER-NAME=shared/owned   (shared value allows more than one cluster to use the subnet)​

- kubernetes.io/role/elb (for public subnet)​

- kubernetes.io/role/internal-elb (for private subnet)